Source/Diagnostics/ReducedDiags/FieldProbe.cpp

```diff
@@ -359,63 +419,112 @@ void FieldProbe::ComputeDiags (int step)
                         part_S[ip] = S; //remember to add lorentz transform
                     }
                 });// ParallelFor Close
-
                 // this check is here because for m_field_probe_integrate == True, we always compute
                 // but we only write when we truly are in an output interval step
-                if (m_intervals.contains(step+1)) {
-                    for (int ip = 0; ip < np; ip++) {
-                        // Fill output array
-                        m_data[ip * noutputs + FieldProbePIdx::Ex] = part_Ex[ip];
-                        m_data[ip * noutputs + FieldProbePIdx::Ey] = part_Ey[ip];
-                        m_data[ip * noutputs + FieldProbePIdx::Ez] = part_Ez[ip];
-                        m_data[ip * noutputs + FieldProbePIdx::Bx] = part_Bx[ip];
-                        m_data[ip * noutputs + FieldProbePIdx::By] = part_By[ip];
-                        m_data[ip * noutputs + FieldProbePIdx::Bz] = part_Bz[ip];
-                        m_data[ip * noutputs + FieldProbePIdx::S] = part_S[ip];
+                if (m_intervals.contains(step+1))
+                {
+                    for (auto ip=0; ip < np; ip++)
+                    {
+                        amrex::ParticleReal xp, yp, zp;
+                        getPosition(ip, xp, yp, zp);
+
+                        // push to output vector
+                        m_data.push_back(xp);
+                        m_data.push_back(yp);
+                        m_data.push_back(zp);
+                        m_data.push_back(part_Ex[ip]);
+                        m_data.push_back(part_Ey[ip]);
+                        m_data.push_back(part_Ez[ip]);
+                        m_data.push_back(part_Bx[ip]);
+                        m_data.push_back(part_By[ip]);
+                        m_data.push_back(part_Bz[ip]);
+                        m_data.push_back(part_S[ip]);
                     }
-                    /* m_data now contains up-to-date values for:
-                     *  [Ex, Ey, Ez, Bx, By, Bz, and S] */
+                /* m_data now contains up-to-date values for:
+                 *  [x, y, z, Ex, Ey, Ez, Bx, By, Bz, and S] */
                 }
-
-                // do we have the one and only probe particle on our MPI rank?
-                if (np > 0)
-                    probe_proc = amrex::ParallelDescriptor::MyProc();
             }
-
         } // end particle iterator loop
-
-        // make sure data is in m_data
         Gpu::synchronize();
+        if (m_intervals.contains(step+1))
+        {
+            // returns total number of mpi notes into mpisize
+            int mpisize;
+            MPI_Comm_size(amrex::ParallelDescriptor::Communicator(), &mpisize);
+
+            // allocates data space for length_array. Will contain size of m_data from each processor
+            int *length_array = NULL;
```

Comment:
More C++-y: use a `std::vector<int>(mpisize, 0)` and pass it as `.data()` to MPI.

If you use the AMReX wrappers, you should be able to pass the vectors directly.